src/main/java/com/yolo/customer/idea/IdeaService.java

```diff
@@ -44,18 +51,31 @@ public class IdeaService {
     @Autowired
     private DietaryRestrictionRepository dietaryRestrictionRepository;
 
+    @Autowired
+    private UserRepository userRepository;
+
 //    @Autowired
 //    private UserRepository userRepository;
 
     public ResponseEntity<Map<String, String>> submitIdeaToVendor(Integer ideaId, String status) {
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = GetContextHolder.getUsernameFromAuthentication(authentication);
+        User loggedInUser = userRepository.findByUsername(username).orElseThrow(() ->
+                new EntityNotFoundException("User with given username does not exist: " + username));
+
         if (status == null || status.isEmpty()) {
-            throw new IllegalArgumentException("Status cannot be empty.");
+            throw new EntityNotFoundException("Status cannot be empty.");
         }
 
         Idea idea = ideaRepository.findById(ideaId)
-                .orElseThrow(() -> new RuntimeException("Idea with ID " + ideaId + " not found"));
+                .orElseThrow(() -> new EntityNotFoundException("Ideas not found"));
+
+        if (!idea.getUserId().equals(loggedInUser.getId())) {
+            throw new RuntimeException("Unauthorized to update idea.");
+        }
 
-        boolean vendorApiSuccess = callVendorApi(idea);
+        boolean vendorApiSuccess = callVendorApi(idea, username);
 
         if (vendorApiSuccess) {
             Long statusId = ideaStatusService.findStatusIdByName(status);
```

Comment:
throw exception if not found

Reply:
Already has exception "EntityNotFoundException("User with given username does not exist")"